cns/nodesubnet/ip_fetcher.go

```diff
@@ -0,0 +1,77 @@
+package nodesubnet
+
+import (
+	"context"
+	"log"
+	"net"
+	"time"
+
+	"github.com/Azure/azure-container-networking/nmagent"
+	"github.com/pkg/errors"
+)
+
+var ErrorRefreshSkipped = errors.New("refresh skipped due to throttling")
+
+// This interface is implemented by the NMAgent Client, and also a mock client for testing
```

Comment:
This will be interpreted as a doc comment. The typical style is a definition with the name of the type as the first word, so:

```suggestion
// InterfaceRetriever is an interface is implemented by the NMAgent Client, and also a mock client for testing.
```